odoo/orm/fields.py

```diff
@@ -1098,9 +1098,16 @@ def update_db_notnull(self, model, column):
             # _init_column may delay computations in post-init phase
             @model.pool.post_init
             def add_not_null():
-                # flush values before adding NOT NULL constraint
-                model.flush_model([self.name])
-                model.pool.post_constraint(apply_required, model, self.name)
+                # At the time this function is called, the model's _fields may have been reset, although
+                # the model's class is still the same. Retrieve the field to see whether the NOT NULL
+                # constraint still applies.
+                field = model._fields[self.name]
+                if not field.required or not field.store:
+                    return
+                # Flush values before adding NOT NULL constraint.
+                # Flush the whole model at once instead of flushing field by field.
+                model.flush_model()
```

Comment:
Why this change ? Is it to force all compute store to be executed ?

Reply:
If you have multiple fields that are not null, you may want to flush the whole table (in one go) and then apply not null on columns instead of (flush column, apply not null, flush other column, ...). This is done in post init, so you should not have any pending computations.

Reply:
> flush column, apply not null, flush other column

AFAIK, flush doesn't work like that :thinking: . If we need to update one field, we will actually updated all dirty field of the same model. See https://github.com/odoo/odoo/blob/1a636a07b3c508d91568a09d1b6ba78690ca4302/odoo/orm/models.py#L6042

Then IMO, this change will actually force any pending update even there isn't none for the current field.